client/my-sites/plugins/marketplace/marketplace-plugin-setup-status/index.tsx

```diff
@@ -14,54 +15,119 @@ import theme from 'calypso/my-sites/plugins/marketplace/theme';
  */
 import 'calypso/my-sites/plugins/marketplace/marketplace-plugin-setup-status/style.scss';
 
 /**
- * This component simulates progress for the purchase flow. It also does any async tasks required in the purchase flow. This includes installing any plugins required.
- * TODO: Refactor component so that it can easily handle multiple speeds of simulated progress
+ * This page busy waits and/or  installs any plugins required in the marketplace purchase flow.
  */
-
 function WrappedMarketplacePluginSetup(): JSX.Element {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
-	const selectedSiteId = useSelector( getSelectedSiteId );
 	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
-	const transferStatus = useSelector( ( state ) =>
-		getAutomatedTransferStatus( state, selectedSiteId )
-	);
-	const { pluginSlugToBeInstalled, isPluginInstalledDuringPurchase } = useSelector(
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const {
+		fetchingStatus: automatedTransferFetchingStatus,
+		status: automatedTransferStatus,
+	} = useSelector( ( state ) => getAutomatedTransfer( state, selectedSiteId ) );
+
+	const { pluginSlugToBeInstalled, siteTransferStatus, pluginInstallationStatus } = useSelector(
 		getPurchaseFlowState
 	);
+	const hasProductSetupError = useSelector( getHasProductSetupError );
+	const isProductSetupComplete = useSelector( getIsProductSetupComplete );
 
-	useEffect( () => {
-		dispatch( fetchAutomatedTransferStatus( selectedSiteId ?? 0 ) );
-	}, [ fetchAutomatedTransferStatus ] );
+	const pluginStatus = useSelector( ( state ) =>
+		getStatusForPlugin( state, selectedSiteId, pluginSlugToBeInstalled )
+	);
+	const isPluginStateLoaded = useSelector( ( state ) => isLoaded( state, [ selectedSiteId ] ) );
+	const isPluginStateFetching = useSelector( ( state ) =>
+		isRequestingForSites( state, [ selectedSiteId ] )
+	);
 
+	// WPorg Plugin Data
+	const isWporgPluginFetching = useSelector( ( state ) =>
+		getIsWporgPluginFetching( state, pluginSlugToBeInstalled )
+	);
+	const isWporgPluginFetched = useSelector( ( state ) =>
+		getIsWporgPluginFetched( state, pluginSlugToBeInstalled )
+	);
+	const wporgPlugin = useSelector( ( state ) => getWporgPlugin( state, pluginSlugToBeInstalled ) );
 	useEffect( () => {
-		if ( pluginSlugToBeInstalled && selectedSiteId ) {
-			dispatch( initiateThemeTransfer( selectedSiteId, null, pluginSlugToBeInstalled ) );
-		} else if ( ! isPluginInstalledDuringPurchase ) {
-			// Invalid State redirect to Yoast marketplace page for now, and maybe a marketplace home view in the future
-			page(
-				`/marketplace/product/details/wordpress-seo/${ selectedSiteSlug }?flags=marketplace-yoast`
-			);
+		if ( ! selectedSiteSlug ) {
+			page( '/home' );
+		} else if ( ! pluginSlugToBeInstalled ) {
+			// A plugin slug should have been provided to reach this page
+			if ( config.isEnabled( 'marketplace-test' ) ) {
+				// eslint-disable-next-line no-console
+				console.error(
```

Comment:
We usually use `debug` with a specific namespace for these. Any reason to go with `console.error()` instead?

Reply:
Oh I haven't used it, how do you use debug and view the messages?

Reply:
Oh is it this util that we use :O
https://github.com/visionmedia/debug

Reply:
In terms of how to use it, you can just search for `'debug'` usage in Calypso's codebase and you'll find hundreds of usages. Essentially usage is like this:

```
// Import it
import debugFactory from 'debug';

// Declare a debugger with our preferred namespace
const debug = debugFactory( 'my-custom-debugging-name' );

// Issue a debugging statement
debug( 'This is an example ' + someVar );
```

Then to view it in the console, you should write this in the console:

```
localStorage.setItem( 'debug', '*' );
```

but since this will display all debug output, it might be too verbose. In that case, you can alter `*` to be the custom namespace you picked above (`my-custom-debugging-name` in the above example).

Let me know if that makes sense 😉

Reply:
Thanks! @tyxla you are awesome 🙇